lib/libutils/ext/include/util.h

```diff
@@ -140,6 +140,41 @@
 #define GENMASK_64(h, l) \
 	(((~UINT64_C(0)) << (l)) & (~UINT64_C(0) >> (64 - 1 - (h))))
 
+/* Get and set bit fields  */
+/* Macros defines for getting and setting bit fields  */
+
+#define GET_FIELD_U32(reg, mask) \
+        (__extension__({ \
+                uint32_t _mask  = (mask); \
+                \
+               ((reg) & _mask) / (_mask & ~SHIFT_U32(_mask, 1)); \
+        }))
+
+#define SET_FIELD_U32(reg, mask, val) \
+        (__extension__({ \
+                uint32_t _mask  = (mask); \
+				uint32_t _val  = (val); \
+				\
+               ((reg & ~_mask) |\
+	           ((_val * (_mask & ~SHIFT_U32(_mask, 1))) & _mask)); \
+	    }))
+
+#define GET_FIELD_U64(reg, mask) \
+        (__extension__({ \
+                uint64_t _mask  = (mask); \
+                \
+               ((reg) & _mask) / (_mask & ~SHIFT_U64(_mask, 1)); \
+        }))
+
+#define SET_FIELD_U64(reg, mask, val) \
+        (__extension__({ \
+                uint64_t _mask  = (mask); \
+				uint64_t _val  = (val); \
+				\
+               ((reg & ~_mask) |\
+	           ((_val * (_mask & ~SHIFT_U64(_mask, 1))) & _mask)); \
+	    }))
+
```

Comment:
You're mixing tabs and spaces :-/

In fact this looks unnecessarily complex... how about a static inline function instead? (and same for the other ones of course).

```C
static inline uint64_t SET_FIELD_U64(uint64_t reg, uint64_t mask, uint64_t val)          
{                                                                               
        return (reg & ~mask) | ((val * (mask & ~SHIFT_U64(mask, 1))) & mask);   
}
```

Reply:
Comment addressed.